### DISCUSSION TITLE
Wireless Tag ZX3D95CE01S-TR Arduino_GFX settings

### DISCUSSION THREAD

**Discussion by ATaimeh** [moononournation/Arduino_GFX#329] [General]
Hello @moononournation,

I hope you are doing well at your end.

Just need to ask, in my case I'm using Wireless Tag ZX3D95CE01S-TR, which I think its the same as ZX3D95CE01S-AR but with different parameters, i have tried to upload the code following all the instructions, and what I got is an empty turned on screen. 

The Arduino_GFX setting code used is:

`#define GFX_BL 5
Arduino_DataBus *bus = new Arduino_SWSPI(
    GFX_NOT_DEFINED /* DC */, 38 /* CS */,
    45 /* SCK or SCLK */, 48 /* MOSI */, GFX_NOT_DEFINED /* MISO */);
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    40 /* DE */, 41 /* VSYNC */, 42 /* HSYNC */, 39 /* PCLK */,
    18 /* R0 */, 8 /* R1 */, 3 /* R2 */, 46 /* R3 */, 10 /* R4 */,
    14 /* G0 */, 13 /* G1 */, 12 /* G2 */, 11 /* G3 */, 16 /* G4 */, 17 /* G5 */,
    45 /* B0 */, 48 /* B1 */, 47 /* B2 */, 0 /* B3 */, 21 /* B4 */,
    1 /* hsync_polarity */, 10 /* hsync_front_porch */, 8 /* hsync_pulse_width */, 50 /* hsync_back_porch */,
    1 /* vsync_polarity */, 10 /* vsync_front_porch */, 8 /* vsync_pulse_width */, 20 /* vsync_back_porch */
    );
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    480 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */,
    bus, GFX_NOT_DEFINED /* RST */, gc9503v_type1_init_operations, sizeof(gc9503v_type1_init_operations));`

I would be grateful if you could help me choosing the right Arduino_GFX settings.

Thanks,
Abdullah

**Comment by moononournation** [comment]
Someone told me TR and AR are different before, but I did not have TR in hand so cannot test it.

**Reply by ATaimeh** [reply]
oh, i just thought its the same, since its the newer  version of AR, well thanks, this info will definitely help, ill continue testing it, and ill reply back once its working.

Thanks.

**Reply by strange-v** [reply]
Hi @ATaimeh, do you have any progress?
Yesterday, I started playing with the same panel but using the LovyanGFX, the result is the same)

**Comment by ATaimeh** [comment]
Hello @strange-v,

Honestly not yet, I'm trying to figure it out, I've checked the RGB panel in the Arduino_GFX library, and did some changes but still no progress. I will surely send the code here once it works, please let me know if you got it working, Thanks!

**Comment by ATaimeh** [comment]
Hello @strange-v,

I think your message was just so lucky to me, it turned on for the first time just now, but colors isn't right, ill fix it and share the code.

**Reply by strange-v** [reply]
Sounds promising)

**Reply by ATaimeh** [reply]
hello @strange-v ,

please try to add this code in GFX settings in .ino file.

`#define GFX_BL 5
Arduino_DataBus *bus = new Arduino_SWSPI(
    GFX_NOT_DEFINED /* DC */, 38 /* CS */,
    45 /* SCK or SCLK */, 48 /* MOSI */, GFX_NOT_DEFINED /* MISO */);
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    40 /* DE */, 41 /* VSYNC */, 42 /* HSYNC */, 39 /* PCLK */,
    18 /* R0 */, 8 /* R1 */, 3 /* R2 */, 46 /* R3 */, 10 /* R4 */,
    14 /* G0 */, 13 /* G1 */, 12 /* G2 */, 11 /* G3 */, 16 /* G4 */, 17 /* G5 */,
    45 /* B0 */, 48 /* B1 */, 47 /* B2 */, 0 /* B3 */, 21 /* B4 */,
    1 /* hsync_polarity */, 8 /* hsync_front_porch */, 10 /* hsync_pulse_width */, 50 /* hsync_back_porch */,
    1 /* vsync_polarity */, 8 /* vsync_front_porch */, 10 /* vsync_pulse_width */, 20 /* vsync_back_porch */);
    // GFX_NOT_DEFINED /*pclk_active_neg*/, GFX_NOT_DEFINED /*prefer_speed*/, false /*useBigEndian*/,
    // /*when de_idle_high is set to 1 panel is giving lines*/ 0 /*de_idle_high*/, 1 /*pclk_idle_high*/);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    480 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush (IPS, in makerfabs project it is IPS) */,
    bus, 41 /* RST */, gc9503v_type1_init_operations, sizeof(gc9503v_type1_init_operations));`

in case you are using squareline studio:

- in project setting -> depth is 16 bit color swap.
-  in lvgl folder -> src -> lv_conf.h, find and set :
   - #define LV_COLOR_16_SWAP   1
   - #define LV_TICK_CUSTOM  1 
   - #if 1 // Set it to "1" to enable content)

hopefully, this will be helpful.

please note the upper code refers to Arduino_GFX library.

**Reply by strange-v** [reply]
It works with Arduino_GFX, though when I apply the same parameters to LovyanGFX, it doesn't.
Thanks for your help!

**Comment by ATaimeh** [comment]
what i noticed is, its about the RST Pin of the display, i didnt try to make it work on lovyaneGFX yet, but if i did ill mention u

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by moononournation** [reply]
ok, committed.

**Comment by Viktortrt** [comment]
thank you guys

**Comment by moononournation** [comment]
If tested ok, please provide a Youtube link of PDQgraphicstest. I will add the dev board declaration to the example.

**Reply by ATaimeh** [reply]
Hello @moononournation 

Sure, Ill take a video and send it to you.

**Reply by ATaimeh** [reply]
Hello @moononournation, 

check this link for the video, https://www.youtube.com/shorts/5u6_C-krK2Q

**Comment by tree1370** [comment]
hi i tested with your help but doesnt work for me. 


>`#include <lvgl.h>
#include <TFT_eSPI.h>
#include <ui.h>

/*Don't forget to set Sketchbook location in File/Preferencesto the path of your UI project (the parent foder of this INO file)*/

/*Change to your screen resolution*/
static const uint16_t TFT_WIDTH  = 480;
static const uint16_t TFT_HEIGHT = 480;

static lv_disp_draw_buf_t draw_buf;
static lv_color_t buf[ TFT_WIDTH * TFT_HEIGHT / 10 ];

TFT_eSPI tft = TFT_eSPI(TFT_WIDTH, TFT_HEIGHT); /* TFT instance */

#if LV_USE_LOG != 0
/* Serial debugging */
void my_print(const char * buf)
{
    Serial.printf(buf);
    Serial.flush();
}
#endif


#define GFX_BL 5
 Arduino_DataBus *bus = new Arduino_SWSPI( GFX_NOT_DEFINED /* DC */, 38 /* CS */, 45 /* SCK or SCLK */, 48 /* MOSI */, GFX_NOT_DEFINED /* MISO */);
  Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel( 40 /* DE */, 41 /* VSYNC */, 42 /* HSYNC */, 39 /* PCLK */, 18 /* R0 */, 8 /* R1 */, 3 /* R2 */, 46 /* R3 */, 10 /* R4 */, 14 /* G0 */, 13 /* G1 */, 12 /* G2 */, 11 /* G3 */, 16 /* G4 */, 17 /* G5 */, 45 /* B0 */, 48 /* B1 */, 47 /* B2 */, 0 /* B3 */, 21 /* B4 */, 1 /* hsync_polarity */, 8 /* hsync_front_porch */, 10 /* hsync_pulse_width */, 50 /* hsync_back_porch */, 1 /* vsync_polarity */, 8 /* vsync_front_porch */, 10 /* vsync_pulse_width */, 20 /* vsync_back_porch */);
   // GFX_NOT_DEFINED /*pclk_active_neg*/, GFX_NOT_DEFINED /*prefer_speed*/, false /*useBigEndian*/, // /*when de_idle_high is set to 1 panel is giving lines*/ 0 /*de_idle_high*/, 1 /*pclk_idle_high*/); 
   Arduino_RGB_Display *gfx = new Arduino_RGB_Display( 480 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush (IPS, in makerfabs project it is IPS) */, bus, 41 /* RST */, gc9503v_type1_init_operations, sizeof(gc9503v_type1_init_operations));

/* Display flushing */
void my_disp_flush( lv_disp_drv_t *disp, const lv_area_t *area, lv_color_t *color_p )
{
    uint32_t w = ( area->x2 - area->x1 + 1 );
    uint32_t h = ( area->y2 - area->y1 + 1 );

    tft.startWrite();
    tft.setAddrWindow( area->x1, area->y1, w, h );
    tft.pushColors( ( uint16_t * )&color_p->full, w * h, true );
    tft.endWrite();

    lv_disp_flush_ready( disp );
}

/*Read the touchpad*/
void my_touchpad_read( lv_indev_drv_t * indev_driver, lv_indev_data_t * data )
{
    uint16_t touchX = 0, touchY = 0;

    bool touched = false;//tft.getTouch( &touchX, &touchY, 600 );

    if( !touched )
    {
        data->state = LV_INDEV_STATE_REL;
    }
    else
    {
        data->state = LV_INDEV_STATE_PR;

        /*Set the coordinates*/
        data->point.x = touchX;
        data->point.y = touchY;

        Serial.print( "Data x " );
        Serial.println( touchX );

        Serial.print( "Data y " );
        Serial.println( touchY );
    }
}

void setup()
{
    Serial.begin( 115200 ); /* prepare for possible serial debug */

    String LVGL_Arduino = "Hello Arduino! ";
    LVGL_Arduino += String('V') + lv_version_major() + "." + lv_version_minor() + "." + lv_version_patch();

    Serial.println( LVGL_Arduino );
    Serial.println( "I am LVGL_Arduino" );

    lv_init();

#if LV_USE_LOG != 0
    lv_log_register_print_cb( my_print ); /* register print function for debugging */
#endif

    tft.begin();          /* TFT init */
    tft.setRotation( 3 ); /* Landscape orientation, flipped */

    lv_disp_draw_buf_init( &draw_buf, buf, NULL, screenWidth * screenHeight / 10 );

    /*Initialize the display*/
    static lv_disp_drv_t disp_drv;
    lv_disp_drv_init( &disp_drv );
    /*Change the following line to your display resolution*/
    disp_drv.hor_res = screenWidth;
    disp_drv.ver_res = screenHeight;
    disp_drv.flush_cb = my_disp_flush;
    disp_drv.draw_buf = &draw_buf;
    lv_disp_drv_register( &disp_drv );

    /*Initialize the (dummy) input device driver*/
    static lv_indev_drv_t indev_drv;
    lv_indev_drv_init( &indev_drv );
    indev_drv.type = LV_INDEV_TYPE_POINTER;
    indev_drv.read_cb = my_touchpad_read;
    lv_indev_drv_register( &indev_drv );


    ui_init();

    Serial.println( "Setup done" );
}

void loop()
{
    lv_timer_handler(); /* let the GUI do its work */
    delay(5);
}
`

**Reply by ATaimeh** [reply]
Hello,

<TFT_eSPI.h> is the library you use, _i didnt test my display in this library_, you need to use <Arduino_GFX> library, since there is different declaration form for display flushing, and setup.

try to use an example from Arduino_GFX library within the parameters mentioned up there and it will work.

Regarding the Squareline Studio, once you export UIs, copy all the UI; such as (ui.h, ui.c, ui.helpers, screen.c, etc..) to your arduino sketch and change the include <//..ui.h> to <ui.h> from all the ui files exported from SLS; to let the sketch read it from the same path of the sketch.

hope this can help.

**Comment by jplehtine** [comment]
There is also a new model, ZX3D95CE01S-V12 with speaker and microphones.
arduino_GFX works with these settings

#include <Arduino_GFX_Library.h>
#define GFX_BL 45
Arduino_DataBus *bus = new Arduino_SWSPI( GFX_NOT_DEFINED, 0, 10, 9, GFX_NOT_DEFINED);
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel( 13 , 12, 11, 14, 10, 9, 40, 20, 19, 41,  46, 3, 42, 8, 18,  2, 17, 16, 1, 15, 1, 8, 10, 50, 1, 8, 10, 20); 
Arduino_GFX *gfx = new Arduino_RGB_Display( 480, 480, rgbpanel, 0, true, bus, 4, gc9503v_type1_init_operations, sizeof(gc9503v_type1_init_operations));

or in esp-idf enviroment

// RGB SPI interface
#define LCD_SPI_DATA0     9  // !< for 1-line SPI, this also refered as MOSI 
#define LCD_SPI_CLK       10 
#define LCD_SPI_CS        0  
#define LCD_SPI_DC        -1
#define LCD_PIN_RST       41  

#define LCD_VSYNC_GPIO    (12) 
#define LCD_HSYNC_GPIO    (11) 
#define LCD_DE_GPIO       (13) 
#define LCD_PCLK_GPIO     (14) 
#define LCD_DATA0_GPIO    (10)  
#define LCD_DATA1_GPIO    (9)  
#define LCD_DATA2_GPIO    (40)  
#define LCD_DATA3_GPIO    (20)  
#define LCD_DATA4_GPIO    (19)  
#define LCD_DATA5_GPIO    (41)  
#define LCD_DATA6_GPIO    (46) 
#define LCD_DATA7_GPIO    (3) 
#define LCD_DATA8_GPIO    (42) 
#define LCD_DATA9_GPIO    (8) 
#define LCD_DATA10_GPIO   (18) 
#define LCD_DATA11_GPIO   (2) 
#define LCD_DATA12_GPIO   (17) 
#define LCD_DATA13_GPIO   (16) 
#define LCD_DATA14_GPIO   (1) 
#define LCD_DATA15_GPIO   (15) 
#define LCD_DISP_EN_GPIO  (-1)

#define LCD_PIN_BK_LIGHT       45

#define LCD_BK_LIGHT_ON_LEVEL  1
#define LCD_BK_LIGHT_OFF_LEVEL !LCD_BK_LIGHT_ON_LEVEL

#define TP_SDA	7 
#define TP_SCL	6
